cEP-0018.md

```diff
@@ -22,6 +22,38 @@ they can write advanced linting bears. This will aim at supporting the more
 flexible visitor based method of AST traversal (as opposed to listener based
 mechanisms).
 
+ANTLR has a `HiddenTokenStream` which recieves all the tokens irrelevant to the
+grammar such as whitespaces. These are not passed on to the parser and as a
+result are not returned when simply fetching text from a node in parse tree.
+
+Consider the code:
+
+```python
+var_x = 5
+if var_x:
+    print('Variable x is', var_x)
+else:
+    print('Variable x is not set')
+```
+
+After converting this to parsetree and fetching source back using
+[this](https://gist.github.com/virresh/643cd1a2d9149668980814d970f98fbc#file-roundtripper-py-L13)
+code we get the following output:
+
+```python
+var_x=5
+ifvar_x:
+print('Variable x is',var_x)
+else:
+print('Variable x is not set')
+```
+
+This prevents us from easily fetching source code.
+
+Since ANTLR has poor support for retrieving original source code from the
```

Comment:
('retrieving' -> 'reconstructing')

It is good to show that round tripping is not possible/feasible.

Please include a visually appealing parse tree of the above example, including any line number and column number information that is provided by ANTLR, so people can see what context is lost.